src/TreeView/TreeView.stories.tsx

```diff
@@ -404,4 +405,148 @@ function TreeItem({
   )
 }
 
+async function wait(ms: number) {
+  return new Promise(resolve => setTimeout(resolve, ms))
+}
+
+async function loadItems(responseTime: number) {
+  await wait(responseTime)
+  return ['Avatar.tsx', 'Button.tsx', 'Checkbox.tsx']
+}
+
+export const AsyncSuccess: Story = args => {
+  const [isLoading, setIsLoading] = React.useState(false)
+  const [asyncItems, setAsyncItems] = React.useState<string[]>([])
+
+  return (
+    <Box sx={{p: 3}}>
+      <nav aria-label="File navigation">
+        <TreeView aria-label="File navigation">
+          <TreeView.Item
+            onExpandedChange={async isExpanded => {
+              if (asyncItems.length === 0 && isExpanded) {
+                // Show loading indicator after a short delay
+                const timeout = setTimeout(() => setIsLoading(true), 500)
```

Comment:
🙋 Is showing the loading item after a short delay an accessible pattern? cc @ericwbailey

Reply:
Context for @ericwbailey: we do this to avoid showing loading indicators before a user might need them. When users see a lot of loading indicators, I think it makes the UI seem slower than it actually is.

Reply:
I think this should be okay.